src/futures/spi.rs

```diff
@@ -0,0 +1,67 @@
+//! Serial Peripheral Interface
+
+use core::future::Future;
+
+/// Async transfer
+pub trait Transfer<W: 'static> {
+    /// Error type
+    type Error;
+
+    /// Associated future for the `transfer` method.
+    type TransferFuture<'a>: Future<Output = Result<(), Self::Error>> + 'a
+    where
+        Self: 'a;
+
+    /// Writes `words` to the slave from the `write` buffer. Puts the words returned in the `read` buffer.
```

Comment:
this should document what happens when the buffers are different size, same as the blocking trait.

Reply:
How does DMA work when the slices would be different sizes? Is that something DMA on MCUs supports?

Reply:
Some hardware supports this out of the box. For example nrf will pad write with zeros or ignore extra read data. Others like stm32 might need impls to emulate it. For example if read is longer than write, start the buf write and when it's done start DMA again writing zeros.

Either way docs should specify whether different sizes are allowed or not, and what the behavior is if they are. #287 is still stuck on discussing this, so I'd just mirror that.